pom.xml

```diff
@@ -10,8 +10,8 @@
     <packaging>pom</packaging>
 
     <properties>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
```

Comment:
This change makes impossible to use Java 8 for the project .
I'm using Java 8 and would like to keep the project Java 8 compatible.

Is it necessary to use Java 11 dropping Java 8 support ?

I think _IF MANIFEST CHANGES WORKS_ for Java 11 this change is not necessary.

Reply:
Kind of agree with you here and that has bee how we have worked with other examples as well, lowest Java version that we happen to need in the actual application code is the optimal.

But, it might be that my mind is changing here. J11 compatibility is more important and as seen here, we can't catch all errors, even if running with J11, if the compilation is set to 8. Also we got two reports from the testing session from developers who had changed this to 11 as the first step before actually doing anything (even though it should not be required, but that's apparently what people are doing9.

Let's see what to do with that part bit later, I'll test the changed manifests actually work with j8.

Reply:
Well, that works other way as well: if we want to have Java 8 support (even if it's not Java 8 based initially) then we should be able to make sure it works with Java 8.

This project works with Java 11 out of the box and it has issues when you _MODIFY_ the project to get Java 11 binaries (force Java 11 compiler).
So I would say it's better to keep it as is.
If we want to make sure that major (or minor) Java version supported then we just need ITs but it will require much more efforts.

Reply:
I was able finally to reproduce the issue via modification pom with Java 11 compiler properties (this is completely missed in the ticket description anyhow).

So it should be fine just exclude the import dependencies from the manifest.

Reply:
It is mentioned in the ticket header and also demonstrated with a separate branch, so I would say it is mentioned in the ticket.

If/as we want to maintain just one branch for this example, I'd say the default target and source version should be decided on which is more important for our evaluators currently: Java 8 or Java 11. Hard to find a definitive answer for that, but at least we are going towards Java 11 (and from the autumn onwards towards Java 17).

I tested with Java 8 (and the bnd hints in this PR, but 1.8 source&target) and that don't seem to break that. So Java 11 upgrade is kind of different than just making this example compatible with Java 11 (not just with JDK 11).

Reply:
This change introduces problems to maintain this branch for me.
So, technically I'm against this change.
But it's up to you to disregard this.

Reply:
No Denis, I'm not disregarding the burden for you that you'd need to change your JAVA_HOME when maintaining this. But this project is not for you or me ;-)